app/controllers/api/base_controller.rb

```diff
@@ -9,6 +9,7 @@ class Api::BaseController < ApplicationController
   skip_before_action :store_current_location
   skip_before_action :require_functional!
 
+  before_action :require_authenticated_user!, if: :disallow_unauthenticated_api_access?
```

Comment:
Unless I'm mistaken, this will break any kind of federation as the ActivityPub-specific controllers (including the inbox controller) inherit from `Api::BaseController`

Reply:
They shouldn't really be inheriting from that, but good catch...